src/components/filters/checkbox.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { FC, ReactNode } from "react"
+
+import { formatNumber } from "../../lib/formattingHelper"
+
+import Checkbox from "../../components/checkbox"
+
+interface Item {
+  value: string | number
+  label: string
+  renderIcon?: () => ReactNode
+}
+
+interface Props {
+  name: string
+  options: Item[]
+  applyFilters: (filtersToApply) => void
+  selected?: (string | number)[]
+  facet?: { [key: string]: number }
+}
+
+const CheckboxFilter: FC<Props> = ({
```

Comment:
I would update the interface to be a bit more generic and not having it fit 1:1 the current implementation.
Something like the following so the option would hold the information and we could ditch the update logic and just update the option that is passed and return it to the callback.

```
interface Item {
  value: string | number
  label: string | () => ReactNode // So it could be flexible to display the icon at the start or end, to have multiple icons etc.
  selected: boolean
  count: number
  labelPosition?: LabelPosition
}

interface Props {
  name: string
  options: Item[]
  onChange: (options) => void
}

```

Reply:
I'll leave it open for now and see how it feels when using it. This looks like an excellent interface for a render prop (to render a single option), but since filters are stateless my concern is that you end up deconstructing and reconstructing the selection all over the place. Potentially this could be refactored later on.